packages/atomic/src/components/facets-v1/facet-search/facet-search-input.tsx

```diff
@@ -0,0 +1,59 @@
+import {FunctionalComponent, h} from '@stencil/core';
+import SearchIcon from 'coveo-styleguide/resources/icons/svg/search.svg';
+import CloseIcon from 'coveo-styleguide/resources/icons/svg/close.svg';
+import {i18n} from 'i18next';
+
+interface FacetSearchInputProps {
+  label: string;
+  query: string;
+  i18n: i18n;
+  onClear(): void;
+  onChange(value: string): void;
+}
+
+export const FacetSearchInput: FunctionalComponent<FacetSearchInputProps> = (
+  props
+) => {
+  const label = props.i18n.t(props.label);
+  const search = props.i18n.t('search');
+  const facetSearch = props.i18n.t('facetSearch', {label});
+  const clear = props.i18n.t('clear');
+  let inputRef: HTMLInputElement | undefined;
+
+  return (
+    <div class="relative h-10 w-full my-3">
+      <input
+        part="search-input"
+        class="w-full h-full border border-neutral rounded px-9 placeholder-neutral-dark text-sm focus:border-primary focus:outline-none"
+        type="text"
+        placeholder={search}
+        aria-label={facetSearch}
+        value={props.query}
+        onInput={(e) =>
+          props.onChange((e.target as HTMLInputElement).value.trim())
```

Comment:
Is the trim needed ? 
In what scenario would it contain something invalid with trailing spaces ?